### PR TITLE
fix: create order event shows close case page when it shouldn't (FPLA-1843)

### PR DESCRIPTION
**Pull request by joez-f**
## JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/FPLA-1843

### Change description ###

The close case page is shown after the user changes their mind regarding whether or not the order applies to all the children. If the order does not apply to all the children then it should not show.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: service/src/test/java/uk/gov/hmcts/reform/fpl/service/GeneratedOrderServiceTest.java [hunk 1]
```diff
@@ -598,114 +598,18 @@ void shouldReturnTrueWhenOrderIsAFinalOrder() {
...
-
-        @Test
-        @DisplayName("Should generate if close case is disabled and all children have final order")
-        void shouldReturnTrueWhenCloseCaseIsDisabledAndAllChildrenHaveFinalOrder() {
-            typeAndDocument = OrderTypeAndDocument.builder().type(CARE_ORDER).subtype(FINAL).build();
-            FurtherDirections directions = FurtherDirections.builder()
-                .directions("And I thought I was mistaken")
-                .build();
-
+        @DisplayName("Should generate a document when generating any other type of order")
```
Comment (lewisbirks) [inline service/src/test/java/uk/gov/hmcts/reform/fpl/service/GeneratedOrderServiceTest.java:607]: minor but the display names can be removed if you want, I had them there as a sort of commenting method for the tests as the amount that were there was getting confusing

File: service/src/test/java/uk/gov/hmcts/reform/fpl/service/GeneratedOrderServiceTest.java [hunk 2]
```diff
@@ -598,114 +598,18 @@ void shouldReturnTrueWhenOrderIsAFinalOrder() {
...
+        void shouldReturnTrueWhenOrderIsNotBlankOrder() {
```
Comment (lewisbirks) [inline service/src/test/java/uk/gov/hmcts/reform/fpl/service/GeneratedOrderServiceTest.java:608]: `shouldReturnTrueWhenOrderIsNotBlankOrderAndFurtherDirectionPresent` or something to that effect. Or you can keep the display name and add it to that

File: service/src/test/java/uk/gov/hmcts/reform/fpl/service/GeneratedOrderServiceTest.java [hunk 3]
```diff
@@ -598,114 +598,18 @@ void shouldReturnTrueWhenOrderIsAFinalOrder() {
...
+            typeAndDocument = OrderTypeAndDocument.builder().type(CARE_ORDER).build();
 
             boolean shouldGenerateDocument = service.shouldGenerateDocument(typeAndDocument,
-                directions,
-                allChildren,
-                null,
-                false);
+                FurtherDirections.builder().build());
```
Comment (lewisbirks) [inline service/src/test/java/uk/gov/hmcts/reform/fpl/service/GeneratedOrderServiceTest.java:612]: no lyrics 😢

File: service/src/test/java/uk/gov/hmcts/reform/fpl/service/GeneratedOrderServiceTest.java (hunk 1 above)
Comment (joez-f) [reply service/src/test/java/uk/gov/hmcts/reform/fpl/service/GeneratedOrderServiceTest.java:607]: sure, will remove them

**Pull request merged**